### PR TITLE
Update rails to 5.2.2

### PR DESCRIPTION
**Pull request by colinbruce**
#### What
Update Rails from 5.1.6 to 5.2.0

#### Why
Rails 6 is nearing release and we should keep more up to date...

#### How
Bump some gems while locking others

--------
#### TODO (wip)
 - [x] Test paperclip interaction in live (this has been superseded by ActiveStorage)
 - [x] Discuss replacing paperclip with ActiveStorage
 - [x] Remove Site Prism deprecation warnings
 - [x] Bump to 5.2.1
 - [x] Bump to 5.2.2

#### TODO (subsequent PRs)
 - [x] Add ticket to jira for migrating paperclip to ActiveStorage
 - [ ] Bump ruby to 6.0
 - [ ] Bump ruby to 6.1

### CODE REVIEW

File: app/models/claims/sort.rb [hunk 1]
```diff
@@ -40,25 +40,25 @@ def sort_field_with_nulls(field, direction)
   end
 
   def sort_last_submitted_at(direction)
-    order(sort_field_with_nulls('last_submitted_at', direction))
+    order(Arel.sql(sort_field_with_nulls('last_submitted_at', direction)))
```
Comment (jsugarman) [inline app/models/claims/sort.rb:43]: can the `Arel.sql` be put into the called method

File: app/models/claims/sort.rb [hunk 2]
```diff
@@ -40,25 +40,25 @@ def sort_field_with_nulls(field, direction)
...
   end
 
   def sort_advocate(direction)
     select('claims.*, ("users"."last_name" || \', \' || "users"."first_name") AS user_name')
       .joins(external_user: :user)
-      .order(sort_field_by('user_name', direction))
-      .order(sort_field_by('created_at', direction))
+      .order(Arel.sql(sort_field_by('user_name', direction)))
```
Comment (jsugarman) [inline app/models/claims/sort.rb]: can the `Arel.sql` be put into the called method

File: spec/controllers/case_workers/admin/case_workers_controller_spec.rb [hunk 3]
```diff
@@ -12,7 +12,7 @@
     before { get :index }
 
     it 'returns http success' do
-      expect(response).to have_http_status(:success)
+      expect(response).to have_http_status(200)
```
Comment (jsugarman) [inline spec/controllers/case_workers/admin/case_workers_controller_spec.rb]: **Not a bocker** As discussed, we sometimes use `be_successful` and sometimes `have_http_status(200)`. could use one or other for consistency

File: app/helpers/application_helper.rb [hunk 4]
```diff
@@ -98,7 +98,7 @@ def column_sort_icon
   end
 
   def dom_id(record, prefix = nil)
-    result = ActionView::RecordIdentifier.dom_id(record, prefix)
+    result = ActionView::RecordIdentifier.dom_id(record, prefix).dup
```
Comment (jsugarman) [inline app/helpers/application_helper.rb:101]: just info: `dup` is clearer but you can also use `+ActionView::RecordIdentifier.dom_id(record, prefix)` to unfreeze

**Review by jsugarman — APPROVED** [review]
👍 Great!. rails 6 here we come. just a few minor suggestions you may already have fixed

File: spec/controllers/case_workers/admin/case_workers_controller_spec.rb (hunk 3 above)
Comment (colinbruce) [reply spec/controllers/case_workers/admin/case_workers_controller_spec.rb]: Switched to use `be_successful` in as many places as I could find and replace!

**Pull request merged**